### PR TITLE
LWDEV-6911 Sell orders are not optimistically updated in the "orders"…

### PR DESCRIPTION
**Pull request by sergeyschegolev**
… tab

### CODE REVIEW

File: src/stores/__tests__/orderListStore.spec.ts [hunk 1]
```diff
@@ -34,24 +34,38 @@ describe('orderList store', () => {
...
         CreateDateTime: '2018-01-17T07:17:40.84Z',
-        Id: '1f4f1673-d7e8-497a-be00-e63cfbdcd0c7',
+        Id: '1f4f1673-d7e8-497a-be00-e63cfbdcd0c6',
         OrderAction: 'Sell',
         Price: 1,
         Status: 'InOrderBook',
         Volume: 0.0001
       });
       expect(orderListStore.allOrders.length).toBe(2);
+      expect(isAdded).toBeTruthy();
```
Comment (optimusway) [inline src/stores/__tests__/orderListStore.spec.ts]: i do not agree
why should it return `boolean`?

you were asking to return newly created `Order` from the API side, but do not do the same thing on the client side

i believe it should return new order list (`Order[]`) and then the caller will decide

File: src/stores/orderListStore.ts [hunk 2]
```diff
@@ -62,15 +63,31 @@ class OrderListStore extends BaseStore {
...
+    this.orders.push(mappers.mapToLimitOrder(addedOrder));
+    return true;
   };
 
   @action
-  deleteOrder = (orderId: string) =>
-    (this.orders = [
-      ...this.orders.filter((order: OrderModel) => order.id !== orderId)
-    ]);
+  deleteOrder = (orderId: string) => {
```
Comment (optimusway) [inline src/stores/orderListStore.ts]: same here for the returning type, it should not be `boolean` i believe

File: src/stores/orderStore.ts [hunk 3]
```diff
@@ -63,7 +64,20 @@ class OrderStore extends BaseStore {
           this.api
             .placeLimit(body)
             // tslint:disable-next-line:no-empty
-            .then(() => {}, this.orderPlacedUnsuccessfully)
+            .then((orderId: any) => {
+              const isAdded = this.rootStore.orderListStore.addOrder({
```
Comment (optimusway) [inline src/stores/orderStore.ts]: one more way to pass a callback to the `addOrder`function, but it's weird

File: src/stores/__tests__/orderListStore.spec.ts (hunk 1 above)
Comment (sergeyschegolev) [reply src/stores/__tests__/orderListStore.spec.ts]: If returning boolean is not suited for this case I think it is better to return single order but not the whole list. The caller should not work with list

**Pull request merged**